### ISSUE TITLE
Importing module openai.api_requestor, but it does not exist

### ISSUE THREAD

**Issue by samirjohnson48** [DataDog/dd-trace-py#8143]
datadog_lambda tries to import openai.api_requestor for tracking openai. However, this module (openai.api_requestor) does not exist (at least, anymore. It probably existed in the previous openai python client). Here is the output log from datadog:

You can reproduce the problem by trying to import openai.api_requestor.

Datadog Lambda Layer Version: public.ecr.aws/datadog/lambda-extension:48
Python version: 3.9

Stacktrace:

```[ERROR]	2024-01-17T22:32:20.807Z		[dd.trace_id=0 dd.span_id=0]	Error No module named 'openai.api_requestor'. Traceback: Traceback (most recent call last):
  File "/var/task/dependencies/datadog_lambda/wrapper.py", line 204, in __init__
    patch_all()
  File "/var/task/dependencies/datadog_lambda/patch.py", line 36, in patch_all
    patch_all_dd()
  File "/var/task/dependencies/ddtrace/_monkey.py", line 203, in patch_all
    patch(raise_errors=False, **modules)
  File "/var/task/dependencies/ddtrace/_monkey.py", line 243, in patch
    when_imported(module)(
  File "/var/task/dependencies/ddtrace/vendor/wrapt/importer.py", line 284, in register
    register_post_import_hook(hook, name)
  File "/var/task/dependencies/ddtrace/vendor/wrapt/decorators.py", line 470, in _synchronized
    return wrapped(*args, **kwargs)
  File "/var/task/dependencies/ddtrace/vendor/wrapt/importer.py", line 82, in register_post_import_hook
    hook(module)
  File "/var/task/dependencies/ddtrace/_monkey.py", line 166, in on_import
    imported_module.patch()
  File "/var/task/dependencies/ddtrace/contrib/openai/patch.py", line 238, in patch
    import openai.api_requestor
ModuleNotFoundError: No module named 'openai.api_requestor'```

**Comment by Yun-Kim** [comment]
Hi @samirjohnson48, thanks for reaching out!

Could you let us know what version of the datadog tracer (ddtrace) and openai you're using? This would help us reproduce your issue. Thanks!

**Comment by samirjohnson48** [comment]
You can reproduce the error also by the following:
<img width="995" alt="Screen Shot 2024-01-24 at 1 46 12 PM" src="https://github.com/DataDog/dd-trace-py/assets/137213736/df3c37f0-a6c5-4223-be44-c95fa4750583">

**Comment by Yun-Kim** [comment]
It looks like this was resolved in `ddtrace==2.2` which added support for `openai>=1.0`. Could you try upgrading to `ddtrace==2.2` or any later version? Thanks!

**Comment by emmettbutler** [comment]
Closed because of the resolution @Yun-Kim mentioned above. Please comment if it doesn't work for you.
